core/domain/classifier_services.py

```diff
@@ -127,6 +127,69 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+
+
+def add_to_training_queue(exploration):
```

Comment:
@seanlip Most of this is fine in restructuring except the part where you said unchanged state names are the set of all states minus the changed state names. Not all states in the exploration are trainable and we don't require classifier mappings for those states in the exploration which aren't even trainable. So, I'll make one function that returns all trainable state names in a dict with two keys: something like
```
trainable_states = {
    state_names_with_changed_answer_groups: [],
    state_names_with_unchanged_answer_groups: []
}
```
Then we can extract what needs to be updated and for what, jobs need to be created. Sounds good?